### ISSUE TITLE
add support for defining environment variables

### ISSUE THREAD

**Issue by NicoK** [erlide/erlide#232]
The erlang preprocessor supports using environment variables in e.g. the "include" directive:
```erlang
-include("$PROJ_ROOT/my_records.hrl")
```
Currently, there is no way of specifying these in the project settings.
I know, it is some sort of advanced preprocessor use but is it possible to add an appropriate field?

http://erlang.org/doc/reference_manual/macros.html

**Comment by vladdu** [comment]
I have to admit that I didn't even know about this feature of the preprocessor... It might be difficult to implement using today's architecture, where the compilation is done in a common backend - different projects might use the same variable with different values.

I think it would be easier when building with make/rebar, which is in the pipeline.

**Comment by NicoK** [comment]
yes - with make or rebar support, I could indeed setup arbitrary configurations and make them work
